mephisto/abstractions/providers/surge_ai/surge_ai_utils.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+Qualification -> Surge AI `Team`
+"""
+
+import os
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+from typing import Union
+
+import surge
+from omegaconf import DictConfig
+from surge import errors as surge_errors
+
+from mephisto.abstractions.crowd_provider import ProviderArgs
+from mephisto.utils.logger_core import get_logger
+
+SURGE_AI_BUDGET = 100000.0
+SURGE_AI_CONFIG_DIR = '~/.surge_ai/'
+SURGE_AI_CONFIG_PATH = os.path.join(SURGE_AI_CONFIG_DIR, 'credentials')
+
+logger = get_logger(name=__name__)
+
+# Surge AI library does not have main exception class (wierd),
+# so we need to catch all their exceptions like this
+surge_exceptions = (
+    surge_errors.SurgeRequestError,
+    surge_errors.SurgeMissingAPIKeyError,
+    surge_errors.SurgeMissingIDError,
+    surge_errors.SurgeMissingAttributeError,
+    surge_errors.SurgeProjectQuestionError,
+    surge_errors.SurgeTaskDataError,
+)
+
+
+def get_surge_ai_api_key() -> Union[str, None]:
+    credentials_path = os.path.expanduser(SURGE_AI_CONFIG_PATH)
+    if os.path.exists(credentials_path):
+        with open(credentials_path, 'r') as f:
+            api_key = f.read().strip()
+            return api_key
+    return None
+
+
+surge.api_key = os.environ.get('SURGE_API_KEY', None) or get_surge_ai_api_key()
+
+
+def check_credentials(*args, **kwargs) -> bool:
+    """Check whether API KEY is correct"""
+    try:
+        # Make a simple request to the API
+        surge.Project.list(page=1)
+        return True
+    except surge_exceptions:
+        return False
+
+
+def setup_credentials(
+    profile_name: str, register_args: Optional[ProviderArgs],
+) -> bool:
+    if not os.path.exists(os.path.expanduser(SURGE_AI_CONFIG_DIR)):
+        os.mkdir(os.path.expanduser(SURGE_AI_CONFIG_DIR))
+
+    with open(os.path.expanduser(SURGE_AI_CONFIG_PATH), 'w') as f:
+        f.write(register_args.api_key)
```

Comment:
similar comment to the Prolific provider, we may need to support multiple concurrent surge accounts.